src/pyspiffe/svid/x509_svid.py

```diff
@@ -1,34 +1,472 @@
+        """
+
+        chain = _parse_der_certificates(certs_chain_bytes)
+        _validate_chain(chain)
+
+        private_key = _parse_der_private_key(private_key_bytes)
+        spiffe_id = _extract_spiffe_id(chain[0])
+
+        return X509Svid(spiffe_id, chain, private_key)
+
+    @classmethod
+    def parse(cls, certs_chain_bytes: bytes, private_key_bytes: bytes) -> 'X509Svid':
+        """Parses the X.509 SVID from PEM blocks containing certificate chain and key
+        bytes.
+
+        The private key must be a PKCS#8 PEM block.
+
+        It is assumed that the leaf certificate is always the first certificate in the parsed chain.
+
+        Args:
+            certs_chain_bytes: Chain of X.509 certificates in PEM format.
+            private_key_bytes: Private key as PKCS#8 PEM block.
+
+        Returns:
+            An instance of a 'X509Svid' containing the chain of certificates, the private key, and the SPIFFE ID of the
+            leaf certificate in the chain.
+
+        Raises:
+            ParseCertificateError: In case the chain of certificates cannot be parsed from the cert_chain_bytes.
+            ParsePrivateKeyError: In case the private key cannot be parsed from the private_key_bytes.
+            InvalidLeafCertificateError: In case the leaf certificate does not have a SPIFFE ID in the URI SAN,
+                                         in case the leaf certificate is CA,
+                                         in case the leaf certificate has 'keyCertSign' as key usage,
+                                         in case the leaf certificate does not have 'digitalSignature' as key usage,
+                                         in case the leaf certificate does not have 'cRLSign' as key usage.
+            InvalidIntermediateCertificateError: In case one of the intermediate certificates is not CA,
+                                                 in case one of the intermediate certificates does not have 'keyCertSign' as key usage.
+        """
+
+        chain = _parse_pem_certificates(certs_chain_bytes)
+
+        _validate_chain(chain)
+
+        private_key = _parse_pem_private_key(private_key_bytes)
+        spiffe_id = _extract_spiffe_id(chain[0])
+
+        return X509Svid(spiffe_id, chain, private_key)
+
+    @classmethod
+    def load(
+        cls,
+        certs_chain_path: str,
+        private_key_path: str,
+        encoding: serialization.Encoding,
+    ) -> 'X509Svid':
+        """Loads the X.509 SVID from PEM or DER encoded files on disk.
+
+        The private key should be without encryption.
+
+        Args:
+            certs_chain_path (str): Path to the file containing one or more X.509 certificates as PEM blocks.
+            private_key_path (str): Path the file containing a private key as PKCS#8 PEM block.
+            encoding (serialization.Encoding): The encoding used to serialize the certs and private key, can be
```

Comment:
Type information should not be part of the docstring here. This is already included on the arguments declaration.
Save method also needs to be update to remove the typing from the docstring.